### ISSUE TITLE
S2688 FN: Do not raise for IsExpression since it works as expected

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6289]
### Description

FN for S2688 (CompareNaN).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
        object[] numbers = { 1, 2, double.NaN };

        if (numbers is [1, 2, double.NaN]) // FN
        {
            Console.WriteLine("Test1");
        }
```

### Expected behavior

The rule should raise an issue, since `numbers is [..,double.NaN]` is implicitly comparing a value with `double.NaN`.

### Actual behavior

The rule does not raise an issue.
